presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpWorkerClient.java

```diff
@@ -261,13 +277,312 @@ public void testResultFetcher()
+                                    }
+                                }
+                            })),
+                taskId,
+                uri,
+                TASK_INFO_JSON_CODEC,
+                PLAN_FRAGMENT_JSON_CODEC,
+                TASK_UPDATE_REQUEST_JSON_CODEC);
+        HttpNativeExecutionTaskResultFetcher taskResultFetcher = new HttpNativeExecutionTaskResultFetcher(
+                newScheduledThreadPool(1),
+                workerClient,
+                Optional.of(new Duration(30, TimeUnit.SECONDS)));
+        CompletableFuture<Void> future = taskResultFetcher.start();
+        try {
+            future.get();
+            List<SerializedPage> pages = new ArrayList<>();
+            Optional<SerializedPage> page = taskResultFetcher.pollPage();
+            while (page.isPresent()) {
+                pages.add(page.get());
+                page = taskResultFetcher.pollPage();
+            }
+
+            assertEquals(numPages, pages.size());
+            for (int i = 0; i < numPages; i++) {
+                assertEquals(pages.get(i).getSizeInBytes(), serializedPageSize);
+            }
+        }
+        catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+            fail();
+        }
+    }
+
+    private static class BreakingLimitResponseManager
+            extends TestingResponseManager.TestingResultResponseManager
+    {
+        private final int serializedPageSize;
+        private final int numPages;
+
+        private int requestCount;
+
+        public BreakingLimitResponseManager(int serializedPageSize, int numPages)
+        {
+            this.serializedPageSize = serializedPageSize;
+            this.numPages = numPages;
+        }
+
+        @Override
+        public Response createResultResponse(String taskId) throws Exception
+        {
+            requestCount++;
+            if (requestCount < numPages) {
+                return createResultResponseHelper(
+                        HttpStatus.OK,
+                        taskId,
+                        requestCount - 1,
+                        requestCount,
+                        false,
+                        serializedPageSize);
+            }
+            else if (requestCount == numPages) {
+                return createResultResponseHelper(
+                        HttpStatus.OK,
+                        taskId,
+                        requestCount - 1,
+                        requestCount,
+                        true,
+                        serializedPageSize);
+            }
+            else {
+                throw new Exception("Retrieving results after buffer completion");
+            }
+        }
+
+        public int getRemainingPageCount()
+        {
+            return numPages - requestCount;
+        }
+    }
+
+    @Test
+    public void testResultFetcherExceedingBufferLimit()
+    {
+        final int numPages = 10;
```

Comment:
we don't use `final` for temp variables; remove `final`

Reply:
Sure. But mind if I ask why?